apollo-ios-pagination/Sources/ApolloPagination/GraphQLQueryPager.swift

```diff
@@ -37,22 +60,27 @@ public class GraphQLQueryPager<InitialQuery: GraphQLQuery, PaginatedQuery: Graph
     client: ApolloClientProtocol,
     initialQuery: InitialQuery,
     extractPageInfo: @escaping (PageExtractionData) -> P,
-    nextPageResolver: @escaping (P) -> PaginatedQuery
+    nextPageResolver: ((P) -> PaginatedQuery)?,
+    previousPageResolver: ((P) -> PaginatedQuery)?
   ) {
     pager = .init(
       client: client,
       initialQuery: initialQuery,
       extractPageInfo: extractPageInfo,
-      nextPageResolver: nextPageResolver
+      nextPageResolver: nextPageResolver,
+      previousPageResolver: previousPageResolver
     )
     Task {
-      let varMapPublisher = await pager.$varMap
+      let previousPageVarMapPublisher = await pager.$previousPageVarMap
+      let nextPageVarMapPublisher = await pager.$nextPageVarMap
       let initialPublisher = await pager.$initialPageResult
-      varMapPublisher.combineLatest(initialPublisher).sink { [weak self] _ in
+      previousPageVarMapPublisher.combineLatest(initialPublisher, nextPageVarMapPublisher).sink { [weak self] _ in
         guard let self else { return }
         Task {
-          let value = await self.pager.pageTransformation()
-          self.canLoadNextSubject.send(value?.canLoadMore ?? false)
+          let canLoadNext = await self.pager.canLoadNext
+          let canLoadPrevious = await self.pager.canLoadPrevious
+          self.canLoadNextSubject.send(canLoadNext)
+          self.canLoadPreviousSubject.send(canLoadPrevious)
         }
```

Comment:
Does this also need a to stored in `cancellables`?

Reply:
It's being stored on line 85, unless I'm misunderstanding?

Reply:
Yupp, I don't know how I missed that. Disregard.